app/main.py

```diff
@@ -1,2 +1,67 @@
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.load_factor = 2 / 3
+        self.current_size = 0
+        self.hash_table = [[] for _ in range(self.capacity)]
+        self.threshold = int(self.capacity * self.load_factor)
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.current_size == self.threshold:
+            self._resize()
+        node = [key, value, hash(key)]
+        index_from_hash = hash(key) % self.capacity
+
+        if not self.hash_table[index_from_hash]:
+            self.hash_table[index_from_hash] = node
+            self.current_size += 1
+            return
+
+        for i in self.hash_table:
+            if i:
+                if i[0] == key:
+                    current_index = self.hash_table.index(i)
+                    self.hash_table[current_index] = node
+                    return
+
+        for index in range(index_from_hash + 1, len(self.hash_table)):
+            if not self.hash_table[index]:
+                self.hash_table[index] = node
+                self.current_size += 1
+                return
+
+        for index in range(0, index_from_hash):
+            if not self.hash_table[index]:
+                self.hash_table[index] = node
+                self.current_size += 1
+                return
```

Comment:
Hello, sorry to bother you but I decided to ask you to redo this task, especially this method. Although it works correctly, the number of lines of code is too large and readability can be improved